pkg/trace/sampler/prioritysampler.go

```diff
@@ -104,43 +104,63 @@ func (s *PrioritySampler) Sample(trace pb.Trace, root *pb.Span, env string) bool
 	if samplingPriority > 1 {
 		return sampled
 	}
+	// short-circuiting root P0 trace chunks that the client passed. The sig is already taken in account
+	// in CountWeightedSig below. This chunk will likely be sampled by the ExceptionSampler
+	if clientDroppedP0s && samplingPriority == 0 && root.ParentID == 0 {
+		return sampled
+	}
 
 	signature := s.catalog.register(ServiceSignature{root.Service, env})
 
 	// Update sampler state by counting this trace
 	s.Sampler.Backend.CountSignature(signature)
 
 	if sampled {
-		s.applyRate(sampled, root, signature)
+		rate := s.applyRate(sampled, root, signature)
 		s.Sampler.Backend.CountSample()
+		// adjust sig score with the expected P0 count for that sig
+		// adjusting this score only matters for root spans
+		if clientDroppedP0s && rate != 0 {
+			// removing 1 to not count twice the P1 chunk
+			weight := 1/rate - 1
+			s.Sampler.Backend.CountWeightedSig(signature, weight)
```

Comment:
nit: I don't know if it really has a chance of happening or what could be the consequences, but we might want to check that we never count a negative weight.